src/main/java/seedu/address/model/assignment/DueDate.java

```diff
@@ -1,29 +1,37 @@
 package seedu.address.model.assignment;
 
 import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.AppUtil.checkArgument;
 
 import java.time.LocalDate;
 import java.time.LocalDateTime;
 import java.time.LocalTime;
 import java.time.format.DateTimeFormatter;
 
 public class DueDate {
-    public static final String MESSAGE_CONSTRAINTS = "Due dates should be in a format dd/MM/yyyy HHmm";
+
+    public static final String MESSAGE_CONSTRAINTS_TIME = "Due date should be in a format d/M/yyyy";
+    public static final String MESSAGE_CONSTRAINTS_DATE = "Due time should be in a format HHmm";
+    public static final String MESSAGE_CONSTRAINTS_DUE_DATE = "Due dates should be in a format d/M/yyyy,HHmm";
 
     protected static final DateTimeFormatter PARSE_DATE_FORMAT = DateTimeFormatter.ofPattern("d/M/yyyy");
     protected static final DateTimeFormatter PARSE_TIME_FORMAT = DateTimeFormatter.ofPattern("HHmm");
     protected static final DateTimeFormatter OUTPUT_FORMAT = DateTimeFormatter.ofPattern("dd MMM yyyy, hh:mm a");
 
-    //TODO: change VALIDATION INDEX
-    //public static final String VALIDATION_REGEX = "\\d{3,}";
+    private static final String DATE_VALIDATION_REGEX =
+            "^([1-9]|[0-2][0-9]|(3)[0-1])(/)([1-9]|((0)[0-9])|((1)[0-2]))(/)\\d{4}$";
+    private static final String TIME_VALIDATION_REGEX = "^(00|[0,1][0-9]|2[0-3])([0-5][0-9])$";
+    private static final String DATE_AND_TIME_VALIDATION_REGEX =
+            "^([1-9]|[0-2][0-9]|(3)[0-1])(/)([1-9]|((0)[0-9])|((1)[0-2]))(/)\\d{4}(,)"
+                    + "(00|[0,1][0-9]|2[0-3])([0-5][0-9])$";
+    private static final String LATEST_TIME_IN_DAY = "2359";
 
-    public final String value;
+    private final String value;
```

Comment:
For the validation regex and value, I think it should be public to keep it consistent with the other classes, also for the value, DueDate.value might be used to display on the UI.

Reply:
yes,good catch